### DISCUSSION TITLE
Two-way sync between Obsidian and CalDAV servers

### DISCUSSION THREAD

**Discussion by davish** [davish/obsidian-full-calendar#347] [Feature Requests]
Generated from #1.

Events are stored as Notes in Obsidian and kept in sync with a selected Google Calendar. This will likely be a pretty involved and large project involving OAuth and the GCal API. Strategies for resolving conflicts between the local Obsidian copy and the Google Calendar copy of an event will need to be fleshed out.

**Comment by barnabaslabs** [comment]
This would be AMAZING!

**Comment by blackpeasy** [comment]
Will buy you lots of coffee if you figure out this one!

**Comment by ahopkins** [comment]
What is current state of GCal sync via oauth?

**Comment by davish** [comment]
Renamed this issue thanks to the research by @dojoteef into the CalDAV protocol.

**Comment by 5drei1** [comment]
+1 this would be amazing 👍

**Comment by datavistics** [comment]
This would really be helpful!

**Comment by Bulwagga** [comment]
This would be extremely useful.  My wife and I share a gardening calendar that is slowly migrating to Obsidian

**Comment by MiracleXYZ** [comment]
+1 for this

**Comment by miseenplace** [comment]
Perhaps this may be helpful? https://github.com/39aldo39/DecSync
Came a across it the other day. Only real device level integration with a CalDav client/server connection is done via android, otherwise you need to run a dav server like Radicle.

**Comment by D4YO** [comment]
+1 for the caldav two way sync !! 😍

**Comment by michaelseidel90** [comment]
@davish I'd love to help you out on this feature.

**Comment by Gr-Nobel** [comment]
is there any plan about remote events. I can't edit google calendar events.

**Comment by FelixBrakel** [comment]
Not sure if you're already aware of this but it seems someone has made a plugin with full CRUD functionality for GCal. 
[https://github.com/YukiGasai/obsidian-google-calendar](https://github.com/YukiGasai/obsidian-google-calendar)
Might be a useful starting point for adding similar features to this plugin.

**Comment by miseenplace** [comment]
I have been experimenting with VdirSyncer (https://github.com/pimutils/vdirsyncer) as a local client that breaks up contacts and events into a file directory with built in database-style management and sync capabilities.

As it directly touches the files, this may be a good option for a desktop system. Full calender would just need to monitor the file directory where a the ics changes would be push/pulled to. If it can work with the multi-ics file structure of vdirsyncer (which I think is better for note-level integration and composability) then a lot of issues associated with file conflicts could be resolved (less potential collisions from the sync side).

**Comment by bepolymathe** [comment]
Any news on this front ?

**Reply by bepolymathe** [reply]
jtx board project interesting too...

**Comment by amikai** [comment]
+1 for this

**Comment by phense** [comment]
+1 yes, please.

**Comment by hexifox** [comment]
Two way sync would be an amazing feature! Just to put it out there, at least for android, Dav5x is used even for bigger things such as nextcloud. Keep up the amazing work 😄

**Comment by kde2podfreebsd** [comment]
Hi! are there any updates about integrate google calendar? @davish

**Comment by komali2** [comment]
USD $100 bounty on this from me. 

I'm also open to exploring a bounty for some kind of VJOURNAL integration.

**Reply by miseenplace** [reply]
Agreed. Also willing to put of funds for this, especially as part of loading from a flat file (or group of flat files directories of ics files).

**Comment by YouFoundJK** [comment]
I have setup GCal integration in my fork [here](https://github.com/YouFoundJK/plugin-full-calendar/releases/latest). It is a two way sync with no local cache for now (you can see, edit and create events on Google Calender - but you need to be online).

If you are interested checkout how to set up the OAuth2.0 [here](https://youfoundjk.github.io/plugin-full-calendar/calendars/gcal/).

**Comment by YouFoundJK** [comment]
The [fork](https://github.com/YouFoundJK/plugin-full-calendar/releases/latest) now have full CalDAV two way sync!
